crates/sui-prover/tests/inputs/quantifiers/count.ok.move

```diff
@@ -0,0 +1,40 @@
+#[allow(unused)]
+module 0x42::quantifiers_count_ok;
+
+#[spec_only]
+use prover::prover::ensures;
+
+#[spec_only]
+use prover::vector_iter::{count, count_range};
+
+#[ext(no_abort)]
+fun x_is_10(x: &u64): bool {
+    *x == 10
+}
+
+#[ext(no_abort)]
+fun x_is_positive(x: &u64): bool {
+    *x > 0
+}
+
+#[ext(no_abort)]
+fun x_is_greater_than_100(x: &u64): bool {
+    *x > 100
+}
+
+#[spec(prove)]
+fun test_count() {
+    let v = vector[10, 20, 10, 30];
+
+    // Test COUNT
+    ensures(count!<u64>(&v, |x| x_is_10(x)) == 2);
+    ensures(count!<u64>(&v, |x| x_is_positive(x)) == 4);
+    ensures(count!<u64>(&v, |x| x_is_greater_than_100(x)) == 0);
+
+    // Test COUNT_RANGE
+    ensures(count_range!<u64>(&v, 0, 4, |x| x_is_10(x)) == 2);
```

Comment:
I notice that examples like this fail if the vector has more than 4 elements.  Maybe that has something to do with the use of `ConcatVec` in the BPL for vectors that big?  I do not have a suggestion for how to fix that, but perhaps the failure on larger vectors can be recorded somewhere.